dipy/io/stateful_tractogram.py

```diff
@@ -548,8 +548,11 @@ def remove_invalid_streamlines(self, epsilon=1e-3):
             indices_to_remove.append(bisect(
                 self._tractogram.streamlines._offsets, i) - 1)
 
-        indices_to_keep = np.setdiff1d(np.arange(len(self._tractogram)),
-                                       np.array(indices_to_remove)).astype(int)
+        indices_to_remove = sorted(set(indices_to_remove))
+
+        indices_to_keep = list(
+            np.setdiff1d(np.arange(len(self._tractogram)),
+                         np.array(indices_to_remove)).astype(int))
```

Comment:
What is the advantage to cast it into `list`?  if it is just styling, I would recommend keeping it as a NumPy array.

Reply:
The existing docstring said that two lists were being returned, which was untrue. One was a list and the other one was a NumPy array. In either case these are 1D structures, and it makes sense/is more consistent to return the same type for both variables. So it makes more sense to return a list rather than changing the docstring. See the commit message in 3be85d0.